mail/mandrill/src/main/java/com/ritense/mail/service/WebhookService.java

```diff
@@ -81,7 +84,7 @@ public void handleMandrillEvents(MandrillWebhookRequest mandrillWebhookRequest)
             if (messageEvent.triggersBlacklisting()) {
                 EmailAddress mailToBlacklist = EmailAddress.from(messageEvent.getMessage().getEmail());
                 String cause = String.format("Mandrill{%s}", messageEvent);
-                blacklistService.blacklist(mailToBlacklist, cause);
+                applicationEventPublisher.publishEvent(new EmailBlacklistedEvent(mailToBlacklist.get(), LocalDateTime.now(), cause));
```

Comment:
Het lijkt me juist logischer om de blacklist service het blacklisten direct te laten doen, en dan een event te emitten zodat andere componenten daar op kunnen reageren.

Reply:
Daar ben ik het niet mee eens. De black list service zit een niveau hoger in de mail module, het lijkt mij daarom beter om hier het event te publishen. Ik wil namelijk voorkomen dat mandril een dependency heeft op de blacklist service. Beide kan maar ik opteer voor deze opzet.

Reply:
Ik vind juist die kant op wel logisch. De mandrill module is een implementatie van de mail feature in valtimo, daarom zou het niet raar zijn om een harde dependency die kant op te hebben. Op zich kan je dit wel via events doen, maar het geeft een beetje rare gevolgen. Wat nou als de blacklist service nieuwe requirements krijg voor het blacklisten van addressen? Wat als het blacklisten niet mag om een of andere reden? andere listeners kunnen dan onterecht al dingen gedaan hebben in de veronderstelling dat een adres al geblacklist is (of gaat worden). Je gebruikt hier een event over iets wat opgetreden is om precies dat ding te bewerkstelligen.